### ISSUE TITLE
[TASK][<LEVEL>] <The Task Title>

### ISSUE THREAD

**Issue by yaooqinn** [apache/kyuubi#5679]
### What's the level of this task?

EASY

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/296) and found no similar tasks.

### Mentor

- [X] I have sufficient expertise on this task, and I volunteer to be a mentor of this task to guide contributors through the task.

### Skill requirements

- scala
- scala-style

### Background and Goals

https://issues.apache.org/jira/browse/SPARK-45338

### Implementation steps

1. apply the rules below to scala linter
```
  <check customId="javaconversions" level="error" class="org.scalastyle.scalariform.TokenChecker" enabled="true">
    <parameters><parameter name="regex">JavaConversions</parameter></parameters>
    <customMessage>Instead of importing implicits in scala.collection.JavaConversions._, import
      scala.jdk.CollectionConverters._ and use .asScala / .asJava methods</customMessage>
  </check>

  <check customId="javaconverters" level="error" class="org.scalastyle.scalariform.TokenChecker" enabled="true">
    <parameters><parameter name="regex">JavaConverters</parameter></parameters>
    <customMessage>Instead of importing implicits in scala.collection.JavaConverters._, import
      scala.jdk.CollectionConverters._ and use .asScala / .asJava methods</customMessage>
  </check>
```

2. fix all failed heckers

### Additional context

Introduction of [2023 Kyuubi Code Contribution Program](https://github.com/apache/kyuubi/issues/5357)

**Comment by pan3793** [comment]
@yaooqinn seems we can not do that, `scala.jdk.CollectionConverters._` is not available in Scala 2.12

<img width="734" alt="image" src="https://github.com/apache/kyuubi/assets/26535726/019572a8-3fb5-4d54-bda8-754f4509bf08">


BTW, PR title is broken

**Comment by yaooqinn** [comment]
OK, I will close this one.
